File: 프로그래머스/lv2/42885. 구명보트/구명보트.py
def solution(people, limit):
    answer = 0
    people.sort()
    l, r = 0, len(people) - 1
    
    while l <= r:
        if people[l] + people[r] <= limit:
            l += 1
        r -= 1
        answer += 1
    
    return answer

File: 프로그래머스/2/42885. 구명보트/구명보트.java
import java.util.Arrays;

class Solution {
    
    public int solution(int[] people, int limit) {
        Arrays.sort(people);
        int answer = 0;
        int l = 0;
        int r = people.length - 1;

        while (l <= r) {
            if (people[l] + people[r] <= limit) {
                l++;
            }
            r--;
            answer++;
        }

        return answer;
    }
}